models/_generate_layers.py

```diff
@@ -345,6 +344,42 @@ def _parse_init_docstring(self):
   def __repr__(self):
     return f"<{self.__class__.__name__} {self.layer_class.__name__} {self.inspect_init_sig}>"
 
+  def get_super_call_code_str(self) -> str:
+    """
+    Inspects the given layer in RETURNN, extracts the super call from there, removes unwanted parameters.
+
+    :return: kwargs_repr
+      Python source code of kwargs for the super() call of the layer which are not excluded.
+      Such that `f"super().__init__({kwargs_repr})"` is valid code.
+    """
+    # get code as string list
+    code = inspect.getsource(self.layer_class.__init__).splitlines()
+
+    # get list of lines starting with super and take the first one
+    lines = [line.strip() for line in code if line.strip().startswith("super")]
+    if not lines:
+      return ''
+
+    # reformat the super call to extract what we need
+    # get the first line which contains super to get the super call
+    super_call = lines[0]  # super_call=super(ChoiceLayer, self).__init__(beam_size=beam_size, search=search, **kwargs)
+    # only keep the part after init of super_call
+    call = super_call.split(".")[1]  # call = "__init__(beam_size=beam_size, search=search, **kwargs)"
+    assert call.startswith("__init__")
+    assert call.endswith(")")
+    # remove "__init__(" and ") from the call string
+    call_pruned = call[len("__init__("):-len(")")]  # call pruned = "beam_size=beam_size, search=search, **kwargs"
+
+    # get list of tuples for parameter with (param_name, value)
+    tup_ls = [x.split("=") for x in call_pruned.split(",") if x.strip() != "**kwargs"]
+    tup_ls = [(key.strip(), value.strip()) for (key, value) in tup_ls]
+    # additionally excluded parameters
+
+    # remove excluded params
```

Comment:
The first comment is outdated.
And I think the second comment also not really needed.
```suggestion
```